packages/maskbook/src/plugins/ITO/UI/ClaimDialog.tsx

```diff
@@ -352,6 +352,7 @@ export function ClaimDialog(props: ClaimDialogProps) {
 
             <SelectERC20TokenDialog
                 disableSearchBar
+                mustIncludeTokens={props.exchangeTokens}
```

Comment:
Strange API design. Why `includeTokens` is not enough?

Reply:
```ts
tokens_ = includeTokens.length
    ? tokens_.filter((x) => includeTokens.some((y) => isSameAddress(y, x.address)))
    : tokens_
```

I suspect is the purpose of `includeTokens` is as a filter? It is easy to misunderstanding anyway. Since `includeTokens` used at many places, I don't have enough time to test them all right now. I decide to make an urgent fix, leave `includeTokens` untouched.

`mustIncludeTokens` means token would certainly be showed on the list. Because we should not strong assume `Services.Ethereum.fetchERC20TokensFromTokenLists()` would fetch successful every time.

Reply:
I got your point. I prefer to use `tokens` as the pre-defined tokens, which will be merged into (or as the replacement of) those tokens from the token lists. The include and exclude tokens as filters like before.

Besides that, we should cache the result of the tokens list in the future according to its version number to reduce network usage. They got a huge data size.